_install-and-configure/upgrade-opensearch/rolling-upgrade.md

```diff
@@ -0,0 +1,212 @@
+   ```bash
+   docker run -d \
+       -p 9201:9200 -p 9601:9600 \
+       -e "discovery.seed_hosts=os-node-01,os-node-02,os-node-03,os-node-04" -e "DISABLE_SECURITY_PLUGIN=true" \
+       -e "DISABLE_INSTALL_DEMO_CONFIG=true" -e "OPENSEARCH_JAVA_OPTS=-Xms512m -Xmx512m" \
+       -e "cluster.name=opensearch-dev-cluster" -e "node.name=os-node-01" \
+       -e "cluster.initial_master_nodes=os-node-01,os-node-02,os-node-03,os-node-04" \
+       -e "bootstrap.memory_lock=true" -e "path.repo=/mnt/snapshots" \
+       --ulimit nofile=65536:65536 --ulimit memlock=-1:-1 \
+       -v data-01:/usr/share/opensearch/data \
+       -v repo-01:/mnt/snapshots \
+       --network opensearch-dev-net \
+       --name os-node-01 \
+       opensearchproject/opensearch:1.3.7
+   ```
+1. Query the `_cat/nodes` endpoint after OpenSearch is running on the new node to confirm that it has joined the cluster.
+   ```bash
+   curl -s "http://localhost:9201/_cat/nodes?v&h=name,version,node.role,master" | column -t
+   ```
+   Sample output:
+   ```bash
+   name        version  node.role  master
+   os-node-02  7.10.2   dimr       *
+   os-node-04  7.10.2   dimr       -
+   os-node-01  7.10.2   dimr       -
+   os-node-03  7.10.2   dimr       -
+   ```
+   In the sample output, the new OpenSearch node reports a running version of `7.10.2` to the cluster. This is the result of `compatibility.override_main_response_version`, which is used when connecting to a cluster with legacy clients that check for a version. You can manually confirm the version of the node by calling the `/_nodes` API endpoint, like the following command. Replace `<nodeName>` with the name of your node. See [Nodes API]({{site.url}}{{site.baseurl}}/api-reference/nodes-apis/index/) to learn more.
+   ```
+   curl -s -X GET 'localhost:9201/_nodes/<nodeName>?pretty=true' | jq -r '.nodes | .[] | "\(.name) v\(.version)"'
+   ```
+   Sample output:
+   ```
+   $ curl -s -X GET 'localhost:9201/_nodes/os-node-01?pretty=true' | jq -r '.nodes | .[] | "\(.name) v\(.version)"'
+   os-node-01 v1.3.7
+   ```
+1. Repeat steps 5 through 9 for each node in your cluster. Remember to upgrade an eligible cluster manager node last. When you are finished replacing the last node, query `_cat/nodes` to confirm that all nodes have joined the cluster. The cluster is now bootstrapped to the new version of OpenSearch.
+   ```bash
+   curl -s "http://localhost:9201/_cat/nodes?v&h=name,version,node.role,master" | column -t
+   ```
+   Sample output:
+   ```bash
+   name        version  node.role  master
+   os-node-04  1.3.7    dimr       -
+   os-node-02  1.3.7    dimr       *
+   os-node-01  1.3.7    dimr       -
+   os-node-03  1.3.7    dimr       -
+   ```
+   There are no longer any eligible cluster manager nodes running the old version, so a cluster manager running the new version was elected. The cluster has now been bootstrapped to the new version.
```

Comment:
Can this description of the consequences of replacing the last node go along with "The cluster is now bootstrapped to the new version of OpenSearch."? Maybe all of it can go together after the sample output. They all describe the condition of the cluster after running the command to confirm all nodes have joined the cluster. Right? In other words, all of the results are these ...

Reply:
Good catch